### ISSUE TITLE
[Firefox] not working windows 10

### ISSUE THREAD

**Issue by stapuft** [Tampermonkey/tampermonkey#807]
(Please fill out the issue template with your details)

## Specifications

- firefox 70.0.1 (64-bit)
- TM: 4.9.6091
- OS: windows 10

## Script

any and all, not even the simplest script will run

**Comment by derjanb** [comment]
This looks good here. Please try a new temporary profile. Is it working then?

https://gofile.io/?c=KFF9FI

**Comment by derjanb** [comment]
Can you please go to "about:debugging#/runtime/this-firefox" and "Inspect" Tampermonkey:

![image](https://user-images.githubusercontent.com/767504/68838097-2c006e00-06be-11ea-93a1-bfd775f0b0ca.png)

Do you see errors being logged? Note: not every error there is Tampermonkey related.

![image](https://user-images.githubusercontent.com/767504/68838250-7681ea80-06be-11ea-9c2e-ef483fd9df5d.png)

**Comment by stapuft** [comment]
nope. i made a new profile, installed tampermonkey, and tried the simplest userscript I could find, that worked up until the latest update, but, it didnt work.
![image](https://user-images.githubusercontent.com/46431157/68838266-343fb580-068c-11ea-8a43-076cf92f1f5c.png)
![image](https://user-images.githubusercontent.com/46431157/68838294-44f02b80-068c-11ea-87a5-fa0126804a19.png)

```
//==UserScript==
// @name Redirect kissanime
// @description Redirect .to to .ru, because the idiots that run the disqus refuse to change the website url.
// @include http://*.kissanime.to/*
// @include https://*.kissanime.to/*
// @include kissanime.to/*
// @run-at document-start
//==/UserScript==


(function () {
	'use strict';
	top.location.hostname = "kissanime.ru";
})();
```

**Comment by derjanb** [comment]
Thanks, but that's an server error and not related to Tampermonkey...

**Comment by stapuft** [comment]
> 
> 
> Thanks, but that's an server error and not related to Tampermonkey...
> 
> Please note, that depending on your location there are Internet issues at the moment:
> https://xn--allestrungen-9ib.de/
> https://downdetector.com/

no its very related, please re-read the post and look at the pictures and code again, the code was supposed to redirect from that 404 page, to the actual site, but it just hangs there on the 404 page.

and yes there are errors, I refreshed a page that a userscript should be running on and got this
![image](https://user-images.githubusercontent.com/46431157/68838803-54bc3f80-068d-11ea-9a61-0eb9c76e6ff9.png)
 which upon clicking takes you here
![image](https://user-images.githubusercontent.com/46431157/68838842-6c93c380-068d-11ea-8358-5e9eca60dfbc.png)

**Comment by derjanb** [comment]
Ah OK, Understood. Sorry. However after changing the script to 

```Javascript 
//==UserScript==
// @name Redirect kissanime
// @description Redirect .to to .ru, because the idiots that run the disqus refuse to change the website url.
// @include http://*.kissanime.to/*
// @include https://*.kissanime.to/*
// @include http://kissanime.to/*
// @run-at document-start
//==/UserScript==


(function () {
	'use strict';
	top.location.hostname = "kissanime.ru";
})();
```

it works fine here. 

Additionally the original script didn't work neither in 4.9.5921 nor in 4.9.6091 (with FF 70.0.1). You can downgrade by installing an older from here: https://addons.mozilla.org/en-US/firefox/addon/tampermonkey/versions/

**Comment by stapuft** [comment]
actually....scrolling up......there were LOADS of errors
![image](https://user-images.githubusercontent.com/46431157/68839458-bf21af80-068e-11ea-83b8-0c736fe85508.png)
![image](https://user-images.githubusercontent.com/46431157/68839484-c9dc4480-068e-11ea-8f23-a385149047ff.png)
![image](https://user-images.githubusercontent.com/46431157/68839497-d3fe4300-068e-11ea-8dc3-662852f256f5.png)
![image](https://user-images.githubusercontent.com/46431157/68839506-d9f42400-068e-11ea-9506-46f0014a13de.png)
![image](https://user-images.githubusercontent.com/46431157/68839518-dfea0500-068e-11ea-8b69-123173deb61e.png)
![image](https://user-images.githubusercontent.com/46431157/68839542-e9736d00-068e-11ea-97c4-80a34e186055.png)
![image](https://user-images.githubusercontent.com/46431157/68839553-eed0b780-068e-11ea-9dba-459f03941c14.png)
![image](https://user-images.githubusercontent.com/46431157/68839571-f728f280-068e-11ea-846d-1bf09696b15f.png)
![image](https://user-images.githubusercontent.com/46431157/68839579-fd1ed380-068e-11ea-8671-621be1d61faf.png)
![image](https://user-images.githubusercontent.com/46431157/68839597-03ad4b00-068f-11ea-87e4-a37837057fdd.png)
![image](https://user-images.githubusercontent.com/46431157/68839602-0b6cef80-068f-11ea-99ce-0fde7c36452b.png)
![image](https://user-images.githubusercontent.com/46431157/68839612-1031a380-068f-11ea-986c-888d10f8ff1b.png)
![image](https://user-images.githubusercontent.com/46431157/68839622-145dc100-068f-11ea-9557-e1f9507550cb.png)
![image](https://user-images.githubusercontent.com/46431157/68839629-16278480-068f-11ea-9c86-d2301b6ea5fa.png)

**Comment by derjanb** [comment]
OK, thanks. `sender.tab is undefined` looks "interesting". As a side note: don't be scared - as I  said many of these messages are not Tampermonkey related. Not sure why the are logged into the extension's console.

Does the script work after my `@include` modification?

**Comment by stapuft** [comment]
> 
> 
> Ah OK, Understood. Sorry. However after changing the script to
> 
> ```js
> //==UserScript==
> // @name Redirect kissanime
> // @description Redirect .to to .ru, because the idiots that run the disqus refuse to change the website url.
> // @include http://*.kissanime.to/*
> // @include https://*.kissanime.to/*
> // @include http://kissanime.to/*
> // @run-at document-start
> //==/UserScript==
> 
> 
> (function () {
> 	'use strict';
> 	top.location.hostname = "kissanime.ru";
> })();
> ```
> 
> it works fine here.
> 
> Additionally the original script didn't work neither in 4.9.5921 nor in 4.9.6091 (with FF 70.0.1). You can downgrade by installing an older from here: https://addons.mozilla.org/en-US/firefox/addon/tampermonkey/versions/

huh? thats odd, it worked for me just fine until today, when every user script stopped working for me, I added the third @include line after it stopped working.....so your change shouldn't have made any difference??

but whats even odder, is me taking your edit, that you said works for you, and having it still not work, in either my normal profile, or the one I just made

**Comment by stapuft** [comment]
> 
> 
> OK, thanks. `sender.tab is undefined` looks "interesting". As a side note: don't be scared - as I said many of these messages are not Tampermonkey related. Not sure why the are logged into the extension's console.
> 
> Does the script work after my `@include` modification?

no, not in either profile.

**Comment by derjanb** [comment]
Another strange thing is what the storage tests prints. I takes 14 seconds to write and get a value to/from Firefox's storage and 17 seconds to delete it.

**Comment by derjanb** [comment]
A lot of these messages are sync related. Maybe you can disable script sync for a better overview.

**Comment by stapuft** [comment]
well i am on a rather crappy computer.
![image](https://user-images.githubusercontent.com/46431157/68840489-d792c980-0690-11ea-85fb-2063e4ea2fd5.png)


disable sync? allright I can try that.

**Comment by stapuft** [comment]
well.....i disabled tampermonkeys sync........but its still running?

**Comment by stapuft** [comment]
nope :(

**Comment by derjanb** [comment]
Can you please find "sender.tab is undefined" and click at "common.js" at the end of the line?

**Comment by stapuft** [comment]
crap ive already restarted, ill get it again tho

**Comment by stapuft** [comment]
![image](https://user-images.githubusercontent.com/46431157/68841523-c34fcc00-0692-11ea-8b0b-958e63828814.png)

**Comment by derjanb** [comment]
OK, this is not a Tampermonkey source file. :-/

Can you please set Tampermonkey's Log Level to "debug", open a new tab, press Ctrl+Shift+J to open the console , enable all log levels

![image](https://user-images.githubusercontent.com/767504/68841799-8bae4780-06c5-11ea-8cd1-2b6aa80e65b6.png)

and then open "http://kissanime.to/"?

**Comment by stapuft** [comment]
![image](https://user-images.githubusercontent.com/46431157/68842199-02cae800-0694-11ea-88b6-6d063b047d21.png)
![image](https://user-images.githubusercontent.com/46431157/68842237-10806d80-0694-11ea-9a44-bc30bb4d0108.png)

**Comment by derjanb** [comment]
Please open the page's console:

![image](https://user-images.githubusercontent.com/767504/68842780-43902480-06c7-11ea-93b6-c5ffefd7ea69.png)

**Comment by derjanb** [comment]
Also, can you please disable all other add-ons for these tests? Thanks.

**Comment by derjanb** [comment]
@stapuft Have you had a chance to check the page console (https://github.com/Tampermonkey/tampermonkey/issues/807#issuecomment-553795764)? Thanks.

**Comment by derjanb** [comment]
@stapuft Does TM BETA 4.10.6100 (https://www.tampermonkey.net/index.php?browser=firefox) work?

**Comment by lucknaumann** [comment]
@derjanb 
I am also having a similar issue with TM & FF on Windows 10 (v.1809) but solved it with Beta (or so it seems so far).

At first, TM wouldn't load any scripts on any pages. FF said it was up to date (v70 - didn't notice a point release) and so did TM. I also have it set to reload pages upon able/disable of scripts but this wasn't working, either. 

In searching for a solution I came upon this thread with your suggestion to install TM Beta (something I was going to do but figured I'd do a quick search for a solution first). 

In attempting to install TM Beta, FF blocked it and didn't show the 'allow this' message, so I went to check my settings. Searching for 'install' showed that FF was actually installing an update (70.0.1)! Had the proper setting anyway and went back to the page, which now showed the 'allow this' message and installed Beta. Restarted FF. 

Upon restart it looked like the scripts (in regular TM) were working (except it still wasn't reloading pages on able/disable of scripts) but on Reddit it acted like I had 'Old Reddit Redirect' enabled, which I didn't (obviously, but I do have 'show disabled scripts' for toggling). Checking that it was indeed TM causing the misbehavior I disabled it and lo and behold! No Old Reddit Redirect. SO -- I enabled Beta and lo and behold! It's working! (Or so it seems for the moment.)

Just wanted to FYI. Hope it helps! (Also wasn't sure I should start another issue. )
